src/middlewared/middlewared/plugins/network.py

```diff
@@ -214,7 +214,7 @@ async def do_update(self, data):
         Discovery support.
         """
         config = await self.config()
-        new_config = config.copy()
+        new_config = data.copy()
```

Comment:
I think we should be doing
```
new_config = config.copy()
new_config.update(data)
```
What this gives us is that during validation we will have clear expectation of keys existing which should be present in the config if maybe we wanted to see what value a field has and based on that validate another field.
Also this is how it's been done in almost all services I think so anybody working would assume we would have the complete config available in `new_config`.